subsystems/drivesubsystem.py

```diff
@@ -459,8 +459,11 @@ def periodic(self):
 
         robotPoseArray = [robotPose.X(), robotPose.Y(), robotPose.rotation().radians()]
 
-        if SmartDashboard.getBoolean(
-            constants.kRobotVisionPoseArrayKeys.validKey, False
+        if (
+            SmartDashboard.getBoolean(
+                constants.kRobotVisionPoseArrayKeys.validKey, False
+            )
+            and not RobotState.isAutonomous()
```

Comment:
Put the `isAutonomous()` check before the SD check since the `isAutonomous()` check is cheaper and the SD check can then be avoided in some situations (the term is "short-circuiting" and most logical operators in most languages support it: https://www.codingninjas.com/codestudio/library/short-circuiting-in-python TLDR: put your cheaper checks earlier)